src/crypto/common.h

```diff
@@ -9,81 +9,81 @@
 #include <config/bitcoin-config.h>
 #endif
 
-#include <stdint.h>
-#include <string.h>
+#include <cstdint>
+#include <cstring>
 
 #include <compat/endian.h>
 
 uint16_t static inline ReadLE16(const unsigned char* ptr)
 {
     uint16_t x;
-    memcpy((char*)&x, ptr, 2);
+    memcpy(reinterpret_cast<char*>(&x), ptr, 2);
```

Comment:
Why casting in the first place since [`memcpy` would reinterpret it as `unsigned char *`](https://en.cppreference.com/w/cpp/string/byte/memcpy) anyways?